### PR TITLE
OS independent Zoom factor & serialization thereof

### LINKED ISSUE #1351: Save UI scale factor
When you zoom in or out in the menu and restart the viewer, the zoom should be restored.

image

### PR DESCRIPTION
**Pull request by Wumpf**
* Command palette actions for zoom in/out/reset
* Zoom factor is now an OS scaling independent factor that is used every frame to determine `pixels_from_points`. This way users can move around the window and it keeps the zoom factor
* show zoom factor in the menu
* Zoom is now serialized out. Fixes #1351

image

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)

### CODE REVIEW

File: crates/re_viewer/src/app.rs [hunk 1]
```diff
@@ -856,6 +875,33 @@ struct AppState {
...
     #[serde(skip)]
     profiler: crate::Profiler,
+
+    /// Zoom factor, independent of OS points_per_pixel setting.
+    ///
+    /// At every frame we check the OS reported scaling (i.e. points_per_pixel)
+    /// and apply this zoom factor to determine the actual points_per_pixel.
+    /// This way, the zooming stays constant when switching between differently scaled screens.
+    /// (Since this is serialized, even between sessions!)
+    zoom_factor: f32,
```
Comment (emilk) [inline crates/re_viewer/src/app.rs:885]: This belongs better in `struct AppOptions`. Then you don't need the big `impl Default for AppState` below.

**Review by emilk — APPROVED** [review]
Works great!

**Review by teh-cmc — DISMISSED** [review]
Code LGTM, but this breaks the web version, at least on Linux (both Firefox & Chromium).

No issue on `main`.

https://user-images.githubusercontent.com/2910679/221929714-77a13e64-a3e0-4c39-a505-3b22d5c95161.mp4

**Comment by Wumpf** [comment]
great catch @teh-cmc! I incorrectly assumed it would not change web behavior and didn't test! 😳

**Comment by Wumpf** [comment]
fixed web!

**Review by emilk — APPROVED** [review]

**Pull request merged**